AK/Slugify.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-2020, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <cctype>
+
+namespace AK {
+
+inline ErrorOr<String> slugify(String const& _raw, char glue = '-')
+{
+    StringBuilder sb;
+    bool just_processed_space = false;
+
+    for (auto const& ch : _raw.bytes()) {
+        if (std::isalnum(ch) != 0) {
+            sb.append(static_cast<char>(std::tolower(ch)));
+            just_processed_space = false;
```

Comment:
This only works with ascii characters (and our version for just ascii would be `is_ascii_alphanumeric(ch)`). Anything non-ascii (including accented characters like `á`) will get skipped over. You probably want to replace `á` with `a` instead? Pinging @trflynn89, does LibUnicode expose a way to do that currently? I can't find anything.

Reply:
I'm having trouble finding a concrete spec for slugification, but my guess would be that we would want to NFD decompose the string. So something like:
```c++
auto decomposed = Unicode::normalize(input, Unicode::NormalizationForm::NFD);
```

If `input` is something like `ábc`, you will get `a\u0301bc`. You can then slugify the string, which will drop the U+0301 accent, and you're left with `abc`.

However that requires linking in LibUnicode, which we cannot do in AK. So uh, `need_for_lib_url++`.

Reply:
I actually want it to support utf-8 string, for example [check this](https://tbhaxor.com/contributions-and-support/#financial-aid-money%F0%9F%92%B0) This is from my website, the emoji on copying serialized to `%F0%9F%92%B0`, but main point is slugification supports utf8.

Reply:
Decomposing a string does not mean it only supports ASCII. In fact, decomposing "💰" does not change it at all:
```c++
auto a = Unicode::normalize("💰"sv, Unicode::NormalizationForm::NFD);
dbgln("{}", a); // Prints "💰"
```

If you're interesting in reading more about normalization, check out [`String.prototype.normalize`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize).

Reply:
I got your point now. 

> However, sometimes more than one code point, or sequence of code points, can represent the same abstract character.

Normalising the string would ensure it is equivalent in binary representation.